pkg/podfacts/podfacts.go

```diff
@@ -1603,6 +1640,22 @@ func (p *PodFacts) QuorumCheckForRestartCluster(restartOnly bool) bool {
 	return restartablePrimaryNodeCount >= (primaryNodeCount+1)/2
 }
 
+// DoesDBHaveQuorum returns true if the cluster will keep quorum
+func (p *PodFacts) DoesDBHaveQuorum(offset int) bool {
+	totalPrimaryCount := 0
+	upPrimaryCount := 0
+	for _, pod := range p.Detail {
+		if !pod.GetIsPrimary() {
+			continue
+		}
+		totalPrimaryCount++
+		if pod.GetUpNode() {
+			upPrimaryCount++
+		}
+	}
+	return 2*(upPrimaryCount-offset) > totalPrimaryCount
```

Comment:
Say you have 2 primaries pri1 pri2 with sizes 4 and 3 so in total 7. you want to shutdown pri2 but first you want to make sure that doing so will not cause quorum loss. So what you will do comparing 2*(7-3) to 7 -> 8 > 7.

Also ">" instead of ">=" because the rule is more than half of the primaries must be up. Correct me if I am wrong.

Reply:
Now I understand this function. The offset is the num of primary nodes to be shut down.

You are right, it should be ">". I verified it in vcluster. I will fix the QuorumCheckForRestartCluster() in my next PR.